features/cellular/framework/AT/AT_CellularContext.cpp

```diff
@@ -687,6 +687,10 @@ nsapi_error_t AT_CellularContext::disconnect()
 {
     tr_info("CellularContext disconnect()");
     if (!_nw || !_is_connected) {
+        if (_new_context_set) {
+            delete_current_context();
+        }
+        _cid = -1;
```

Comment:
delete_current_context() sets _cid to -1 only in successful case. What is the meaning of this line?

Reply:
Disconnect must be set `_cid = -1` to not expect any `_cid` to be valid/preferred on the next connect request, even when we did not create that `_cid` ourselves.